In [ ]:
print("test")

test


Apa Itu Klasifikasi ? <br>
Klasifikasi adalah proses untuk mengindenfikasi dari model untuk menghasilkan data sebutan dari proses training

# LOAD DATA

In [ ]:
from sklearn.datasets import load_iris
iris_datasets = load_iris()

In [ ]:
print("jenis isian dataset iris + \n {}".format(iris_datasets.keys()))

jenis isian dataset iris + 
 dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'])


In [ ]:
print("target name = {}".format(iris_datasets['target_names']))

target name = ['setosa' 'versicolor' 'virginica']


In [ ]:
print("feature name = {}".format(iris_datasets['feature_names']))

feature name = ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']


In [ ]:
print("Data =\n {}".format(iris_datasets['data']))

Data =
 [[5.1 3.5 1.4 0.2]
 [4.9 3.  1.4 0.2]
 [4.7 3.2 1.3 0.2]
 [4.6 3.1 1.5 0.2]
 [5.  3.6 1.4 0.2]
 [5.4 3.9 1.7 0.4]
 [4.6 3.4 1.4 0.3]
 [5.  3.4 1.5 0.2]
 [4.4 2.9 1.4 0.2]
 [4.9 3.1 1.5 0.1]
 [5.4 3.7 1.5 0.2]
 [4.8 3.4 1.6 0.2]
 [4.8 3.  1.4 0.1]
 [4.3 3.  1.1 0.1]
 [5.8 4.  1.2 0.2]
 [5.7 4.4 1.5 0.4]
 [5.4 3.9 1.3 0.4]
 [5.1 3.5 1.4 0.3]
 [5.7 3.8 1.7 0.3]
 [5.1 3.8 1.5 0.3]
 [5.4 3.4 1.7 0.2]
 [5.1 3.7 1.5 0.4]
 [4.6 3.6 1.  0.2]
 [5.1 3.3 1.7 0.5]
 [4.8 3.4 1.9 0.2]
 [5.  3.  1.6 0.2]
 [5.  3.4 1.6 0.4]
 [5.2 3.5 1.5 0.2]
 [5.2 3.4 1.4 0.2]
 [4.7 3.2 1.6 0.2]
 [4.8 3.1 1.6 0.2]
 [5.4 3.4 1.5 0.4]
 [5.2 4.1 1.5 0.1]
 [5.5 4.2 1.4 0.2]
 [4.9 3.1 1.5 0.2]
 [5.  3.2 1.2 0.2]
 [5.5 3.5 1.3 0.2]
 [4.9 3.6 1.4 0.1]
 [4.4 3.  1.3 0.2]
 [5.1 3.4 1.5 0.2]
 [5.  3.5 1.3 0.3]
 [4.5 2.3 1.3 0.3]
 [4.4 3.2 1.3 0.2]
 [5.  3.5 1.6 0.6]
 [5.1 3.8 1.9 0.4]
 [4.8 3.  1.4 0.3]
 [5.1 3.8 1.6 0.2]
 [4.6 3.2 1.4 0.2]
 [5.3 3.7 1.5 0.2]
 [5.  3.3 1.4 0.2]
 [7.  3.2 4.7 1.4]
 [6.4 3.2 4.5 1.5]
 [6.

In [ ]:
print("Data =\n {}".format(iris_datasets['data'][:5]))

Data =
 [[5.1 3.5 1.4 0.2]
 [4.9 3.  1.4 0.2]
 [4.7 3.2 1.3 0.2]
 [4.6 3.1 1.5 0.2]
 [5.  3.6 1.4 0.2]]


In [ ]:
print("Target = \n{}".format(iris_datasets['target'])) #jenis bunga iris
# 0 = setosa
# 1 = versicolor
# 2 = virginica

Target = 
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2 2
 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2
 2 2]


In [ ]:
#melihat tipe data
print("Tipe data = {}".format(type(iris_datasets['data'])))
print("Tipe target = {}".format(type(iris_datasets['target'])))

Tipe data = <class 'numpy.ndarray'>
Tipe target = <class 'numpy.ndarray'>


In [ ]:
print("Ukuran Data = {}".format(iris_datasets['data'].shape))
print("Ukuran Target = {}".format(iris_datasets['target'].shape))

Ukuran Data = (150, 4)
Ukuran Target = (150,)


# **Proses Training dan Testing**

**Membagi 150 Data menjadi data untuk training dan data untuk testing**<Br>
**Dicoba membagi data training (75%) dan data testing (25%)**

In [ ]:
from sklearn.model_selection import train_test_split
#x = data, y = target
x_train, x_test, y_train, y_test = train_test_split(
    iris_datasets['data'],iris_datasets['target'], test_size = 0.25, random_state = 0)
#random_state = 0 -> kalau di run ulang, isi data yang dibagi tetap sama

In [ ]:
#jumlah data training
print("x_train shape : {} ".format(x_train.shape))#DATA = ( Sepal Length, sepal widt, petal length, petal widt)
print("y_train shape : {} ".format(y_train.shape))#TARGET = ( Sentosa, versicolor, verginical)

x_train shape : (112, 4) 
y_train shape : (112,) 


In [ ]:
#jumlah data training
print("x_test shape : {} ".format(x_test.shape))#DATA = ( Sepal Length, sepal widt, petal length, petal widt)
print("y_test shape : {} ".format(y_test.shape))#TARGET = ( Sentosa, versicolor, verginical)

x_test shape : (38, 4) 
y_test shape : (38,) 


# **PROSES TRAINING**

In [ ]:
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors = 1)

In [ ]:
knn.fit(x_train, y_train)

KNeighborsClassifier(n_neighbors=1)

melakukan prediksi

In [ ]:
import numpy as np
x_baru = np.array([[5,2.1, 1, 0.2]])
prediksi = knn.predict(x_baru)
print("HASIL PREDIKSI KNN = {}".format(prediksi))
print("HASIL PREDIKSI NAMA BUNGA = {}".format(iris_datasets['target_names'][prediksi]))

HASIL PREDIKSI KNN = [0]
HASIL PREDIKSI NAMA BUNGA = ['setosa']


In [ ]:
import numpy as np
x_baru = np.array([[3,5.1, 5, 4.2]])
prediksi = knn.predict(x_baru)
print("HASIL PREDIKSI KNN = {}".format(prediksi))
print("HASIL PREDIKSI NAMA BUNGA = {}".format(iris_datasets['target_names'][prediksi]))

HASIL PREDIKSI KNN = [2]
HASIL PREDIKSI NAMA BUNGA = ['virginica']


# EVALUASI MODEL

UKUR AKURASI

In [ ]:
y_pred = knn.predict(x_test)
print(y_pred)

[2 1 0 2 0 2 0 1 1 1 2 1 1 1 1 0 1 1 0 0 2 1 0 0 2 0 0 1 1 0 2 1 0 2 2 1 0
 2]


In [ ]:
print("HASIL AKURAASI MODEL = {}".format(knn.score(x_test,y_test)))
#hasil akurasi 97%

HASIL AKURAASI MODEL = 0.9736842105263158


TRAININGNYA 90% TES NYA 10%

In [ ]:
from sklearn.model_selection import train_test_split
#x = data, y = target
x_train, x_test, y_train, y_test = train_test_split(
    iris_datasets['data'],iris_datasets['target'], test_size = 0.75, random_state = 0)
#random_state = 0 -> kalau di run ulang, isi data yang dibagi tetap sama

In [ ]:
print("HASIL AKURAASI MODEL = {}".format(knn.score(x_test,y_test)))

HASIL AKURAASI MODEL = 0.9911504424778761


# TUGAS <Br>
TRAINING 75% TEST 25%

In [ ]:
from sklearn.datasets import load_breast_cancer
breast_cancer_dataset = load_breast_cancer()

In [ ]:
print("jenis isian dataset breast cancer + \n {}".format(breast_cancer_dataset.keys()))

jenis isian dataset breast cancer + 
 dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'])


In [ ]:
print("target name = {}".format(breast_cancer_dataset['target_names']))

target name = ['malignant' 'benign']


In [ ]:
print("feature name = {}".format(breast_cancer_dataset['feature_names']))

feature name = ['mean radius' 'mean texture' 'mean perimeter' 'mean area'
 'mean smoothness' 'mean compactness' 'mean concavity'
 'mean concave points' 'mean symmetry' 'mean fractal dimension'
 'radius error' 'texture error' 'perimeter error' 'area error'
 'smoothness error' 'compactness error' 'concavity error'
 'concave points error' 'symmetry error' 'fractal dimension error'
 'worst radius' 'worst texture' 'worst perimeter' 'worst area'
 'worst smoothness' 'worst compactness' 'worst concavity'
 'worst concave points' 'worst symmetry' 'worst fractal dimension']


In [ ]:
print("Data =\n {}".format(breast_cancer_dataset['data']))

Data =
 [[1.799e+01 1.038e+01 1.228e+02 ... 2.654e-01 4.601e-01 1.189e-01]
 [2.057e+01 1.777e+01 1.329e+02 ... 1.860e-01 2.750e-01 8.902e-02]
 [1.969e+01 2.125e+01 1.300e+02 ... 2.430e-01 3.613e-01 8.758e-02]
 ...
 [1.660e+01 2.808e+01 1.083e+02 ... 1.418e-01 2.218e-01 7.820e-02]
 [2.060e+01 2.933e+01 1.401e+02 ... 2.650e-01 4.087e-01 1.240e-01]
 [7.760e+00 2.454e+01 4.792e+01 ... 0.000e+00 2.871e-01 7.039e-02]]


In [ ]:
print("Data =\n {}".format(breast_cancer_dataset['data'][:5]))

Data =
 [[1.799e+01 1.038e+01 1.228e+02 1.001e+03 1.184e-01 2.776e-01 3.001e-01
  1.471e-01 2.419e-01 7.871e-02 1.095e+00 9.053e-01 8.589e+00 1.534e+02
  6.399e-03 4.904e-02 5.373e-02 1.587e-02 3.003e-02 6.193e-03 2.538e+01
  1.733e+01 1.846e+02 2.019e+03 1.622e-01 6.656e-01 7.119e-01 2.654e-01
  4.601e-01 1.189e-01]
 [2.057e+01 1.777e+01 1.329e+02 1.326e+03 8.474e-02 7.864e-02 8.690e-02
  7.017e-02 1.812e-01 5.667e-02 5.435e-01 7.339e-01 3.398e+00 7.408e+01
  5.225e-03 1.308e-02 1.860e-02 1.340e-02 1.389e-02 3.532e-03 2.499e+01
  2.341e+01 1.588e+02 1.956e+03 1.238e-01 1.866e-01 2.416e-01 1.860e-01
  2.750e-01 8.902e-02]
 [1.969e+01 2.125e+01 1.300e+02 1.203e+03 1.096e-01 1.599e-01 1.974e-01
  1.279e-01 2.069e-01 5.999e-02 7.456e-01 7.869e-01 4.585e+00 9.403e+01
  6.150e-03 4.006e-02 3.832e-02 2.058e-02 2.250e-02 4.571e-03 2.357e+01
  2.553e+01 1.525e+02 1.709e+03 1.444e-01 4.245e-01 4.504e-01 2.430e-01
  3.613e-01 8.758e-02]
 [1.142e+01 2.038e+01 7.758e+01 3.861e+02 1.425e-01 2.839e-

In [ ]:
print("Target = \n{}".format(breast_cancer_dataset['target']))

Target = 
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 1 0 0 0 0 0 0 0 0 1 0 1 1 1 1 1 0 0 1 0 0 1 1 1 1 0 1 0 0 1 1 1 1 0 1 0 0
 1 0 1 0 0 1 1 1 0 0 1 0 0 0 1 1 1 0 1 1 0 0 1 1 1 0 0 1 1 1 1 0 1 1 0 1 1
 1 1 1 1 1 1 0 0 0 1 0 0 1 1 1 0 0 1 0 1 0 0 1 0 0 1 1 0 1 1 0 1 1 1 1 0 1
 1 1 1 1 1 1 1 1 0 1 1 1 1 0 0 1 0 1 1 0 0 1 1 0 0 1 1 1 1 0 1 1 0 0 0 1 0
 1 0 1 1 1 0 1 1 0 0 1 0 0 0 0 1 0 0 0 1 0 1 0 1 1 0 1 0 0 0 0 1 1 0 0 1 1
 1 0 1 1 1 1 1 0 0 1 1 0 1 1 0 0 1 0 1 1 1 1 0 1 1 1 1 1 0 1 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 1 1 1 1 1 1 0 1 0 1 1 0 1 1 0 1 0 0 1 1 1 1 1 1 1 1 1 1 1 1
 1 0 1 1 0 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 0 1 0 1 1 1 1 0 0 0 1 1
 1 1 0 1 0 1 0 1 1 1 0 1 1 1 1 1 1 1 0 0 0 1 1 1 1 1 1 1 1 1 1 1 0 0 1 0 0
 0 1 0 0 1 1 1 1 1 0 1 1 1 1 1 0 1 1 1 0 1 1 0 0 1 1 1 1 1 1 0 1 1 1 1 1 1
 1 0 1 1 1 1 1 0 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 0 1 0 1 1 1 1 1 0 1 1
 0 1 0 1 1 0 1 0 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 0 1
 1 1 1 1 1 1 0 

In [ ]:
#melihat tipe data
print("Tipe data = {}".format(type(breast_cancer_dataset['data'])))
print("Tipe target = {}".format(type(breast_cancer_dataset['target'])))

Tipe data = <class 'numpy.ndarray'>
Tipe target = <class 'numpy.ndarray'>


In [ ]:
print("Ukuran Data = {}".format(breast_cancer_dataset['data'].shape))
print("Ukuran Target = {}".format(breast_cancer_dataset['target'].shape))

Ukuran Data = (569, 30)
Ukuran Target = (569,)


Membagi 150 Data menjadi data untuk training dan data untuk testing
Dicoba membagi data training (75%) dan data testing (25%)

In [ ]:
from sklearn.model_selection import train_test_split
#x = data, y = target
z_train, z_test, a_train, a_test = train_test_split(
    breast_cancer_dataset['data'],breast_cancer_dataset['target'], test_size = 0.25, random_state = 0)
#random_state = 0 -> kalau di run ulang, isi data yang dibagi tetap sama

In [ ]:
#jumlah data training
print("z_train shape : {} ".format(z_train.shape))
print("a_train shape : {} ".format(a_train.shape))

z_train shape : (426, 30) 
a_train shape : (426,) 


In [ ]:
#jumlah data training
print("z_test shape : {} ".format(z_test.shape))#DATA = ( Sepal Length, sepal widt, petal length, petal widt)
print("a_test shape : {} ".format(a_test.shape))#TARGET = ( Sentosa, versicolor, verginical)

z_test shape : (143, 30) 
a_test shape : (143,) 


In [ ]:
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors = 1)

In [ ]:
knn.fit(z_train, a_train)

KNeighborsClassifier(n_neighbors=1)

In [ ]:
import numpy as np
z_baru = np.array([[15, 20, 100, 0.2, 0.3, 1, 5, 0.1, 0.2, 0.1, 1, 2, 3, 0.2, 0.1, 4, 5, 6, 0.3, 0.2, 10, 11, 12, 0.4, 0.5, 6, 7, 0.3, 0.2, 0.1]])
prediksi = knn.predict(x_baru)
print("HASIL PREDIKSI KNN = {}".format(prediksi))
print("HASIL PREDIKSI NAMA BUNGA = {}".format(breast_cancer_dataset['target_names'][prediksi]))

HASIL PREDIKSI KNN = [1]
HASIL PREDIKSI NAMA BUNGA = ['benign']


In [ ]:
a_pred = knn.predict(x_test)
print(a_pred)

[1 1 1 0 1 1 1 1 1 1 0 1 1 1 0 0 1 0 0 0 0 1 1 1 0 1 1 1 1 0 1 0 1 0 1 0 1
 0 1 0 0 1 0 1 0 0 1 1 1 0 0 1 0 1 1 1 1 1 1 0 0 0 1 1 0 1 0 0 0 1 1 0 1 1
 0 1 1 1 1 1 0 0 0 1 0 1 1 1 0 0 1 0 1 0 1 1 0 1 1 1 1 1 1 1 0 1 0 1 0 1 1
 0 0 1 1 1 0 1 1 1 1 1 1 1 0 1 1 1 1 1 0 1 1 1 1 1 1 0 0 1 1 1 0]


In [ ]:
print("HASIL AKURAASI MODEL = {}".format(knn.score(z_test,a_test)))
#hasil akurasi 91%

HASIL AKURAASI MODEL = 0.916083916083916
